### ISSUE TITLE
model.eval()

### ISSUE THREAD

**Issue by JaejinCho** [pytorch/tutorials#2083]
In the tutorial below, isn't it better to have model.eval() for more general cases in addition to the context manager torch.no_grad(), or at least have a brief explanation regarding the difference between the two? I think no_grad does not take care of dropout or batchnorm. Although not having model.eval() is fine in this tutorial, it seems necessary generally for evaluation.

https://github.com/pytorch/tutorials/blob/515227009df5d71f2aa766859184ce675f4c62db/beginner_source/basics/optimization_tutorial.py#L172

cc @suraj813 @jerryzh168 @z-a-f @vkuzo

**Comment by z-a-f** [comment]
@svekars I believe `label:arch-optimization` is for `quantization`/`sparsity` related topics, so might not be applicable for this issue.

**Comment by zabboud** [comment]
Would the following additions to the tutorial be sufficient? 

Current test loop:
```python
def test_loop(dataloader, model, loss_fn):
    size = len(dataloader.dataset)
    num_batches = len(dataloader)
    test_loss, correct = 0, 0

    with torch.no_grad():
        for X, y in dataloader:
            pred = model(X)
            test_loss += loss_fn(pred, y).item()
            correct += (pred.argmax(1) == y).type(torch.float).sum().item()

    test_loss /= num_batches
    correct /= size
    print(f"Test Error: \n Accuracy: {(100*correct):>0.1f}%, Avg loss: {test_loss:>8f} \n")
```

Addition of comments and model.eval()

```python
def test_loop(dataloader, model, loss_fn):
    # Set the model to evaluation mode - important for batch normalization and dropout layers
    # Unnecessary in this situation but added for best practices
    model.eval()
    size = len(dataloader.dataset)
    num_batches = len(dataloader)
    test_loss, correct = 0, 0

    # Evaluating the model with torch.no_grad() ensures that no gradients are computed during test mode
    # also serves to reduce unnecessary gradient computations and memory usage for tensors with requires_grad=True
    with torch.no_grad():
        for X, y in dataloader:
            pred = model(X)
            test_loss += loss_fn(pred, y).item()
            correct += (pred.argmax(1) == y).type(torch.float).sum().item()
```
